content/inference/pytorch.md

```diff
@@ -0,0 +1,410 @@
+
+#### GPU Support
+Tensors are created on the host CPU by default:
+```python
+b = torch.zeros([2,3], dtype=torch.int32)
+b.device
+>>> cpu
+```
+
+You can also create tensors on any available GPUs:
+```python
+torch.cuda.is_available() # check that a GPU is available
+>>> True 
+cuda0 = torch.device('cuda:0')
+c = torch.ones([2,3], dtype=torch.int32, device=cuda0)
+c.device
+>>> cuda:0
+```
+
+You can also move tensors between devices:
+```python
+b = b.to(cuda0)
+b.device
+>>> cuda:0
+```
+
+There are trade-offs between computations on the CPU and GPU. GPUs have limited memory and there is a cost associated with transfering data from CPUs to GPUs. However, GPUs perform heavy matrix operations much faster than CPUs, and are therefore often used to speed up training routines.  
+
+```python
+N = 1000 # 
+for i, N in enumerate([10, 100, 500, 1000, 5000]):
+    print("({},{}) Matrices:".format(N,N))
+    M1_cpu = torch.randn(size=(N,N), device='cpu')
+    M2_cpu = torch.randn(size=(N,N), device='cpu')
+    M1_gpu = torch.randn(size=(N,N), device=cuda0)
+    M2_gpu = torch.randn(size=(N,N), device=cuda0)
+    if (i==0):
+        print('Check devices for each tensor:')
+        print('M1_cpu, M2_cpu devices:', M1_cpu.device, M2_cpu.device)
+        print('M1_gpu, M2_gpu devices:', M1_gpu.device, M2_gpu.device)
+
+    def large_matrix_multiply(M1, M2):
+        return M1 * M2.transpose(0,1)
+    
+    n_iter = 1000
+    t_cpu = Timer(lambda: large_matrix_multiply(M1_cpu, M2_cpu))
+    cpu_time = t_cpu.timeit(number=n_iter)/n_iter
+    print('cpu time per call: {:.6f} s'.format(cpu_time))
+
+    t_gpu = Timer(lambda: large_matrix_multiply(M1_gpu, M2_gpu))
+    gpu_time = t_gpu.timeit(number=n_iter)/n_iter
+    print('gpu time per call: {:.6f} s'.format(gpu_time))
+    print('gpu_time/cpu_time: {:.6f}\n'.format(gpu_time/cpu_time))
+
+>>> (10,10) Matrices:
+Check devices for each tensor:
+M1_cpu, M2_cpu devices: cpu cpu
+M1_gpu, M2_gpu devices: cuda:0 cuda:0
+cpu time per call: 0.000008 s
+gpu time per call: 0.000015 s
+gpu_time/cpu_time: 1.904711
+
+(100,100) Matrices:
+cpu time per call: 0.000015 s
+gpu time per call: 0.000015 s
+gpu_time/cpu_time: 0.993163
+
+(500,500) Matrices:
+cpu time per call: 0.000058 s
+gpu time per call: 0.000016 s
+gpu_time/cpu_time: 0.267371
+
+(1000,1000) Matrices:
+cpu time per call: 0.000170 s
+gpu time per call: 0.000015 s
+gpu_time/cpu_time: 0.089784
+
+(5000,5000) Matrices:
+cpu time per call: 0.025083 s
+gpu time per call: 0.000011 s
+gpu_time/cpu_time: 0.000419
+```
+
+The complete list of Torch Tensor operations is available in the [docs](https://pytorch.org/docs/stable/torch.html?highlight=mm). 
+
+#### Autograd
+
+Backpropagation occurs automatically through autograd. For example, consider the following function and its derivatives:
+\begin{align} 
```

Comment:
Will this be rendered properly once merged? It's not being rendered on the GitHub markdown preview, but maybe it will be by mkdocs.